File: 24-03-2025_Count_days_without_meetings.py
def countDays(self, days: int, meetings: List[List[int]]) -> int:
    meetings.sort()
    r = 0
    ans = days

    for pair in meetings:
        if pair[1] <= r:
            continue
        else:
            ans -= (pair[1] - max(pair[0], r))
            if r < pair[0]:
                ans -= 1
            r = pair[1]
    
    return ans

File: 24-03-2025_Count_days_without_meetings.java
public int countDays(int days, int[][] meetings) {
    Arrays.sort(meetings, (a, b) -> Integer.compare(a[0], b[0]));
    int r = 0;
    int ans = days;
    
    for (int[] pair : meetings) {
        if (pair[1] <= r) {
            continue;
        } else {
            ans -= (pair[1] - Math.max(pair[0], r));
            if (r < pair[0]) {
                ans--;
            }
            r = pair[1];
        }
    }
    
    return ans;
}
